packages/celotool/src/e2e-tests/slashing_tests.ts

```diff
@@ -0,0 +1,249 @@
+      const lockedGold = await kit.contracts.getLockedGold()
+
+      const validatorsContract = await kit._web3Contracts.getValidators()
+      const history = await validatorsContract.methods.getMembershipHistory(signer).call()
+      const historyIndex = history[0].length - 1
+
+      await slasher.methods
+        .slash(
+          blockNumber + 12,
+          4,
+          4,
+          historyIndex,
+          [],
+          [],
+          [],
+          [NULL_ADDRESS],
+          [NULL_ADDRESS],
+          [0]
+        )
+        .send({ from: validator, gas: 5000000 })
+
+      const balance = await lockedGold.getAccountTotalLockedGold(signer)
+      assert.equal(balance.toString(10), '9000000000000000000000')
+    })
+  })
+
+  describe('test slashing for double signing', () => {
+    before(async function(this: any) {
+      this.timeout(0) // Disable test timeout
+      await restart()
+    })
+
+    it('slash for double signing', async function(this: any) {
+      this.timeout(0) // Disable test timeout
+      const slasher = await kit._web3Contracts.getDoubleSigningSlasher()
+
+      await waitUntilBlock(doubleSigningBlock.number)
+
+      const other = headerFromBlock(web3, doubleSigningBlock)
+
+      const num = await slasher.methods.getBlockNumberFromHeader(other).call()
+
+      const header = headerFromBlock(web3, await web3.eth.getBlock(num))
+
+      const bitmap1 = await slasher.methods.getVerifiedSealBitmapFromHeader(header).call()
+      const bitmap2 = await slasher.methods.getVerifiedSealBitmapFromHeader(other).call()
+
+      let bmNum1 = new BigNumber(bitmap1).toNumber()
+      let bmNum2 = new BigNumber(bitmap2).toNumber()
+      bmNum1 = bmNum1 >> 1
+      bmNum2 = bmNum2 >> 1
+      let signerIdx = 1
+      for (let i = 1; i < 5; i++) {
+        if ((bmNum1 & 1) === 1 && (bmNum2 & 1) === 1) {
+          break
+        }
+        signerIdx++
+        bmNum1 = bmNum1 >> 1
+        bmNum2 = bmNum2 >> 1
+      }
+
+      const signer = await slasher.methods.validatorSignerAddressFromSet(signerIdx, num).call()
+      const validator = (await kit.web3.eth.getAccounts())[0]
+      await kit.web3.eth.personal.unlockAccount(validator, '', 1000000)
+
+      const lockedGold = await kit.contracts.getLockedGold()
+      const validatorsContract = await kit._web3Contracts.getValidators()
+      const history = await validatorsContract.methods.getMembershipHistory(signer).call()
+      const historyIndex = history[0].length - 1
+
+      await slasher.methods
+        .slash(
+          signer,
+          signerIdx,
+          header,
+          other,
+          historyIndex,
+          [],
+          [],
+          [],
+          [NULL_ADDRESS],
+          [NULL_ADDRESS],
+          [0]
+        )
+        .send({ from: validator, gas: 5000000 })
+
+      const balance = await lockedGold.getAccountTotalLockedGold(signer)
+      assert.equal(balance.toString(10), '9000000000000000000000')
```

Comment:
nit: Please include a comment as to why we expect this particular balance (ditto for the other test)